### ISSUE TITLE
[Feature request] Ability to span multiple tiles with shortcuts too

### ISSUE THREAD

**Issue by hrbtk** [domferr/tilingshell#45]
It would be cool to be able to do what you can do now with CTRL + ALT + mouse but without mouse (maybe SUPER + ALT + Arrows?). 

If for example I have layout like this (screenshot) and I want 2 windows side by side, now I have to expand right window down with mouse, even if I used ex-Gnome shortcuts SUPER + Arrows to place left window

![Screenshot from 2024-06-23 22-35-04](https://github.com/domferr/tilingshell/assets/44986423/4347ee65-7e30-469c-aba2-8fdb8253cd7d)

WDYT?

**Comment by domferr** [comment]
Hey! Thank you for your suggestion! 
Since theis extension extends GNOME's default shortcuts <kbd>SUPER</kbd> + <kbd>Arrows</kbd> I think it would be cool to implement your idea!
Something like this: if you selected <kbd>ALT</kbd> as a key for spanning multiple tiles, then you can use <kbd>SUPER</kbd> + <kbd>ALT</kbd> + <kbd>Arrows</kbd> to span the focused window (instead of moving it).

**Comment by ebits21** [comment]
This shortcut would complete that basics of what I would want from the extension. Great idea.

I often want quarter tiles but sometimes want half tiles. Either expanding across tiles (alt works great as mentioned) or a shortcut to change the active layout (also would be great) could accomplish that.

**Comment by domferr** [comment]
Hey, I've a quick update for you. 
I made some testing, but I got some doubts. Let's say you selected <kbd>ALT</kbd> as a key for spanning multiple tiles, then I was thinking about automatically picking <kbd>SUPER</kbd> + <kbd>ALT</kbd> + <kbd>Arrows</kbd> to span the focused window (instead of moving it) by default. Despite being customizable, the problem I got is that the keybinding may be already used: since it is picked automatically, everything breaks without even knowing ahah!

**Comment by ebits21** [comment]
> Hey, I've a quick update for you. 
> 
> I made some testing, but I got some doubts. Let's say you selected <kbd>ALT</kbd> as a key for spanning multiple tiles, then I was thinking about automatically picking <kbd>SUPER</kbd> + <kbd>ALT</kbd> + <kbd>Arrows</kbd> to span the focused window (instead of moving it) by default. Despite being customizable, the problem I got is that the keybinding may be already used: since it is picked automatically, everything breaks without even knowing ahah!
> 
> 
> 
> I think the best solution would be to avoid any default or automatic selection of this. The user will have to go to the preferences and select the preferred keybindings

Makes sense.

**Comment by domferr** [comment]
Hey guys, here we go! This is going to be released with the next update of Tiling Shell (v13). If you want to try it in advance and give some feedback, I leave here the files. Enjoy!

- GNOME shell version 42, 43 or 44: [GNOME.42-44.tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/16550098/GNOME.42-44.tilingshell%40ferrarodomenico.com.zip)
- GNOME shell version 45 or 46: [tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/16550099/tilingshell%40ferrarodomenico.com.zip)

**Comment by ebits21** [comment]
Checked it out. Works great! 

I used Alt + Arrows to expand as it wouldn't let me use the Super + Alt + Arrows combination.

**Comment by kaesetoast** [comment]
Thanks for the dope feature, this was the last piece I needed for it to become my new default :-)

I just installed this and the spanning works great. I noticed another bug though, not sure if it is related to this feature. But it definitely was'n there before. For some reason I can't move a window to the rightmost corner any more. It works on the first try but then stops working. I can still move windows to the left and the middle with shortcuts, but for the right it just won't do it. Still works via mouse though.

https://github.com/user-attachments/assets/0a053bf8-0d91-4d5e-9071-259cd7528565

Changing the layout seems to reset it. If I change the layout, first try works again but every subsequent try fails again.
